test/specs/modules/Dropdown/Dropdown.spec.js

```diff
@@ -54,4 +113,217 @@ describe('Dropdown', () => {
     expect(options.at(0).props().text).to.equal('foo');
     expect(options.at(1).props().text).to.equal('bar');
   });
+
+  it('should have icons, flags and images in option', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        options: [
+          {
+            text: 'foo',
+            value: 1,
+            flag: 'cn',
+            icon: 'question',
+            image: {
+              src: '/test',
+            },
+          },
+        ],
+      },
+    });
+    const item = wrapper.find(DropdownItem);
+    const icon = item.find(Icon);
+    expect(icon.is(Icon)).to.equal(true);
+    const flag = item.find(Flag);
+    expect(flag.is(Flag)).to.equal(true);
+    const image = item.find(Image);
+    expect(image.is(Image)).to.equal(true);
+  });
+
+  it('should choose option', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        options: [{ text: 'foo', value: 1 }, { text: 'bar', value: 2 }],
+      },
+    });
+
+    const options = wrapper.findAll(DropdownItem);
+    options.at(0).trigger('click');
+    expect(wrapper.emitted('input')[0][0]).to.equal(1);
+    options.at(1).trigger('click');
+    expect(wrapper.emitted('input')[1][0]).to.equal(2);
+  });
+
+  it('should choose few options', () => {
+    const wrapper = shallow(DropdownWithRequired, {
+      propsData: {
+        multiple: true,
+        options: [{ text: 'foo', value: 1 }, { text: 'bar', value: 2 }],
+      },
+    });
+
+    wrapper.vm.$on('input', (value) => {
+      wrapper.setProps({ value });
+    });
```

Comment:
Do we need this?

Reply:
You don't need to check that v-model works, as it's already well tested by vue itself. Want you want to know is if the `input` event is fired with the right value:

You can remove both the `vm.$on` and write (haven't tested myself):

```js
const options = wrapper.findAll(DropdownItem);
options.at(0).trigger('click');
expect(wrapper.emitted().input[0][0]).to.equal([1]);
wrapper.setProps({ value: [1] });
options.at(0).trigger('click');
expect(wrapper.emitted().input[0][1]).to.equal([1, 2]);
```